pip/req/req_set.py

```diff
@@ -249,11 +249,10 @@ def add_requirement(self, install_req, parent_req_name=None):
                     self.requirement_aliases[name.lower()] = name
                 result = [install_req]
             else:
-                if not existing_req.constraint:
-                    # No need to scan, we've already encountered this for
-                    # scanning.
-                    result = []
-                elif not install_req.constraint:
+                # Assume there's no need to scan, and that we've already
+                # encountered this for scanning.
+                result = []
+                if not install_req.constraint:
```

Comment:
I'm wondering if it should not be `if not install_req.constraint and existing_req.constraint:` since it was an elif...